geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedQueryRunner.scala

```diff
@@ -254,6 +266,16 @@ class MergedQueryRunner(ds: HasGeoMesaStats, stores: Seq[(Queryable, Option[Filt
       super.getReturnSft(sft, hints)
     }
   }
+
+  private def doParallelScan(
+      readers: Seq[FeatureReader[SimpleFeatureType, SimpleFeature]],
+      single: FeatureReader[SimpleFeatureType, SimpleFeature] => CloseableIterator[SimpleFeature]): CloseableIterator[SimpleFeature] = {
+    if (parallel) {
+      SelfClosingIterator(readers.par.map(single).iterator).flatMap(i => i)
```

Comment:
Do the readers need to be closed?  Does this do that?

Reply:
i was hoping you'd confirm haha. i _think_ the flatMap(i => i) will take care of it.

Reply:
`flatten` seemed a bit more complicated to implement for closeable iterators

Reply:
Honestly, this stuff is complicated enough that I'd end up writing a unit test to convince myself that I didn't screw up...

And later, I'd find that I missed something.:)

Reply:
added some tests

Reply:
Ship it when ready!  (Or tell me to push the button.)